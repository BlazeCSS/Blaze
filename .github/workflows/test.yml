name: CI

on:
  push:
    branches:
      - master

jobs:
  build:
    name: Install, test and build
    
    runs-on: ubuntu-latest
    
    steps:
      - name: Checkout repo
        uses: actions/checkout@master

      - name: Install dependencies
        run: npm ci

      - name: Test
        run: npm run test

      - name: Build site
        run: npm run build
      
      - name: Switch to site dir
        run: cd ./packages/site

      - name: Deploy site
        uses: w9jds/firebase-action@master
        with:
          args: deploy --only hosting
        env:
          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
          PROJECT_ID: "default"
