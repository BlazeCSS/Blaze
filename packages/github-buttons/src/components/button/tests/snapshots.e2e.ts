import { snapIt } from '../../../test';

const component = 'github-button';
const snap = snapIt(component);

snap('emtpy', `<github-button></github-button>`);
snap('default options', `<github-button user="BlazeUI" repo="blaze"></github-button>`);
snap(
  'all the options',
  `<github-button user="BlazeUI" repo="blaze" type="follow" text="Join us" large count></github-button>`
);
