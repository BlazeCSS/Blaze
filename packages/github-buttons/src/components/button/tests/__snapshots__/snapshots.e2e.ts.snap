// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all the options 1`] = `"<style data-style-tag=\\"github-button\\">.c-github-button{display:inline-block;overflow:hidden;font-size:0;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;white-space:nowrap;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.c-github-button__link{display:inline-block;height:18px;padding:0 5px;color:#24292e;font-weight:600;line-height:18px;text-decoration:none;vertical-align:bottom;border:1px solid #d1d2d3;border-radius:.25em;outline:0;cursor:pointer}.c-github-button__link:focus{border-color:#c8e1ff}.c-github-button__link--button{font-size:11px;background-color:#eff3f6;background-image:-webkit-gradient(linear,left top, left bottom,from(#fafbfc),to(#e4ebf0));background-image:linear-gradient(180deg,#fafbfc,#e4ebf0);background-repeat:repeat-x;background-size:110% 110%}.c-github-button__link--button:active{background-color:#e9ecef;background-image:none;border-color:#afb1b2;-webkit-box-shadow:inset 0 .15em .3em rgba(27,31,35,.15);box-shadow:inset 0 .15em .3em rgba(27,31,35,.15)}.c-github-button__link--button:hover{background-color:#e6ebf1;background-image:-webkit-gradient(linear,left top, left bottom,from(#f0f3f6),to(#dce3ec));background-image:linear-gradient(180deg,#f0f3f6,#dce3ec);border-color:#afb1b2}.c-github-button__link--count{position:relative;margin-left:5px;font-size:10px;background-color:#fff}.c-github-button__link--count:hover{color:#0366d6}.c-github-button__link--count b,.c-github-button__link--count i{position:absolute;top:50%;left:0;display:block;width:0;height:0;margin:-4px 0 0 -4px;border:4px solid transparent;border-left-width:0}.c-github-button__link--count b{border-right-color:#d1d2d3}.c-github-button__link--count:focus b{border-right-color:#c8e1ff}.c-github-button__link--count i{margin-left:-3px;border-right-color:#fff}.c-github-button__icon{position:relative;top:2px;display:inline-block;height:14px;vertical-align:top;fill:currentColor}.u-large .c-github-button__link{height:26px;line-height:26px}.u-large .c-github-button__link--button{padding:0 10px;font-size:12px}.u-large .c-github-button__link--count{margin-left:7px;padding:0 7px;font-size:11px}.u-large .c-github-button__link--count b,.u-large .c-github-button__link--count i{margin:-5px 0 0 -5px;border-width:5px 5px 5px 0}.u-large .c-github-button__link--count i{margin-left:-4px}.u-large .c-github-button__icon{top:4px;height:18px}.c-github-button__label{text-transform:capitalize}</style><div class=\\"c-github-button u-large\\"><a href=\\"https://github.com/BlazeUI/blaze\\" target=\\"_blank\\" class=\\"c-github-button__link c-github-button__link--button\\" aria-label=\\"Open BlazeUI/blaze on GitHub\\"><svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewbox=\\"0 0 16 16\\" class=\\"c-github-button__icon c-github-button__icon--mark-github\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z\\"></path></svg><span class=\\"c-github-button__label\\">Join us</span></a></div>"`;

exports[`default options 1`] = `"<style data-style-tag=\\"github-button\\">.c-github-button{display:inline-block;overflow:hidden;font-size:0;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;white-space:nowrap;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.c-github-button__link{display:inline-block;height:18px;padding:0 5px;color:#24292e;font-weight:600;line-height:18px;text-decoration:none;vertical-align:bottom;border:1px solid #d1d2d3;border-radius:.25em;outline:0;cursor:pointer}.c-github-button__link:focus{border-color:#c8e1ff}.c-github-button__link--button{font-size:11px;background-color:#eff3f6;background-image:-webkit-gradient(linear,left top, left bottom,from(#fafbfc),to(#e4ebf0));background-image:linear-gradient(180deg,#fafbfc,#e4ebf0);background-repeat:repeat-x;background-size:110% 110%}.c-github-button__link--button:active{background-color:#e9ecef;background-image:none;border-color:#afb1b2;-webkit-box-shadow:inset 0 .15em .3em rgba(27,31,35,.15);box-shadow:inset 0 .15em .3em rgba(27,31,35,.15)}.c-github-button__link--button:hover{background-color:#e6ebf1;background-image:-webkit-gradient(linear,left top, left bottom,from(#f0f3f6),to(#dce3ec));background-image:linear-gradient(180deg,#f0f3f6,#dce3ec);border-color:#afb1b2}.c-github-button__link--count{position:relative;margin-left:5px;font-size:10px;background-color:#fff}.c-github-button__link--count:hover{color:#0366d6}.c-github-button__link--count b,.c-github-button__link--count i{position:absolute;top:50%;left:0;display:block;width:0;height:0;margin:-4px 0 0 -4px;border:4px solid transparent;border-left-width:0}.c-github-button__link--count b{border-right-color:#d1d2d3}.c-github-button__link--count:focus b{border-right-color:#c8e1ff}.c-github-button__link--count i{margin-left:-3px;border-right-color:#fff}.c-github-button__icon{position:relative;top:2px;display:inline-block;height:14px;vertical-align:top;fill:currentColor}.u-large .c-github-button__link{height:26px;line-height:26px}.u-large .c-github-button__link--button{padding:0 10px;font-size:12px}.u-large .c-github-button__link--count{margin-left:7px;padding:0 7px;font-size:11px}.u-large .c-github-button__link--count b,.u-large .c-github-button__link--count i{margin:-5px 0 0 -5px;border-width:5px 5px 5px 0}.u-large .c-github-button__link--count i{margin-left:-4px}.u-large .c-github-button__icon{top:4px;height:18px}.c-github-button__label{text-transform:capitalize}</style><div class=\\"c-github-button\\"><a href=\\"https://github.com/BlazeUI/blaze\\" target=\\"_blank\\" class=\\"c-github-button__link c-github-button__link--button\\" aria-label=\\"Open BlazeUI/blaze on GitHub\\"><svg version=\\"1.1\\" width=\\"14\\" height=\\"16\\" viewbox=\\"0 0 14 16\\" class=\\"c-github-button__icon c-github-button__icon--star\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z\\"></path></svg><span class=\\"c-github-button__label\\">star</span></a></div>"`;

exports[`emtpy 1`] = `"<style data-style-tag=\\"github-button\\">.c-github-button{display:inline-block;overflow:hidden;font-size:0;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;white-space:nowrap;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.c-github-button__link{display:inline-block;height:18px;padding:0 5px;color:#24292e;font-weight:600;line-height:18px;text-decoration:none;vertical-align:bottom;border:1px solid #d1d2d3;border-radius:.25em;outline:0;cursor:pointer}.c-github-button__link:focus{border-color:#c8e1ff}.c-github-button__link--button{font-size:11px;background-color:#eff3f6;background-image:-webkit-gradient(linear,left top, left bottom,from(#fafbfc),to(#e4ebf0));background-image:linear-gradient(180deg,#fafbfc,#e4ebf0);background-repeat:repeat-x;background-size:110% 110%}.c-github-button__link--button:active{background-color:#e9ecef;background-image:none;border-color:#afb1b2;-webkit-box-shadow:inset 0 .15em .3em rgba(27,31,35,.15);box-shadow:inset 0 .15em .3em rgba(27,31,35,.15)}.c-github-button__link--button:hover{background-color:#e6ebf1;background-image:-webkit-gradient(linear,left top, left bottom,from(#f0f3f6),to(#dce3ec));background-image:linear-gradient(180deg,#f0f3f6,#dce3ec);border-color:#afb1b2}.c-github-button__link--count{position:relative;margin-left:5px;font-size:10px;background-color:#fff}.c-github-button__link--count:hover{color:#0366d6}.c-github-button__link--count b,.c-github-button__link--count i{position:absolute;top:50%;left:0;display:block;width:0;height:0;margin:-4px 0 0 -4px;border:4px solid transparent;border-left-width:0}.c-github-button__link--count b{border-right-color:#d1d2d3}.c-github-button__link--count:focus b{border-right-color:#c8e1ff}.c-github-button__link--count i{margin-left:-3px;border-right-color:#fff}.c-github-button__icon{position:relative;top:2px;display:inline-block;height:14px;vertical-align:top;fill:currentColor}.u-large .c-github-button__link{height:26px;line-height:26px}.u-large .c-github-button__link--button{padding:0 10px;font-size:12px}.u-large .c-github-button__link--count{margin-left:7px;padding:0 7px;font-size:11px}.u-large .c-github-button__link--count b,.u-large .c-github-button__link--count i{margin:-5px 0 0 -5px;border-width:5px 5px 5px 0}.u-large .c-github-button__link--count i{margin-left:-4px}.u-large .c-github-button__icon{top:4px;height:18px}.c-github-button__label{text-transform:capitalize}</style><div class=\\"c-github-button\\"><a href=\\"https://github.com/undefined/undefined\\" target=\\"_blank\\" class=\\"c-github-button__link c-github-button__link--button\\" aria-label=\\"Open undefined/undefined on GitHub\\"><svg version=\\"1.1\\" width=\\"14\\" height=\\"16\\" viewbox=\\"0 0 14 16\\" class=\\"c-github-button__icon c-github-button__icon--star\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z\\"></path></svg><span class=\\"c-github-button__label\\">star</span></a></div>"`;
